code/datums/status_effects/xeno_buffs.dm

```diff
@@ -871,3 +871,499 @@
+
+// ***************************************
+// *********** Upgrade Chambers Buffs - Utility
+// ***************************************
+/atom/movable/screen/alert/status_effect/upgrade_toxin
+	name = "Toxin"
+	desc = "Inject toxin on attack."
+	icon_state = "xenobuff_generic"
+
+/atom/movable/screen/alert/status_effect/upgrade_toxin/Click()
+	var/datum/status_effect/upgrade_toxin/status_effect = attached_effect
+	var/datum/reagent/toxin/toxin_choice = show_radial_menu(status_effect.buff_owner, status_effect.buff_owner, GLOB.defiler_toxin_images_list, radius = 35, require_near = TRUE)
+	if(!toxin_choice)
+		return
+	for(var/defiler_toxin in GLOB.defiler_toxin_type_list)
+		var/datum/reagent/reagent = GLOB.chemical_reagents_list[defiler_toxin]
+		if(reagent.name == toxin_choice)
+			status_effect.selected_reagent = reagent.type
+			break
+	status_effect.buff_owner.balloon_alert(status_effect.buff_owner, "[toxin_choice]")
+
+/datum/status_effect/upgrade_toxin
+	id = "upgrade_toxin"
+	duration = -1
+	status_type = STATUS_EFFECT_UNIQUE
+	alert_type = /atom/movable/screen/alert/status_effect/upgrade_toxin
+	/// Owner typed as an xenomorph.
+	var/mob/living/carbon/xenomorph/buff_owner
+	/// The amount of toxins to inject.
+	var/toxin_amount_per_chamber = 1
+	/// The amount of times to multiply the toxin amount by.
+	var/chamber_scaling = 0
+	/// Currently selected reagent to inject.
+	var/datum/reagent/toxin/selected_reagent = /datum/reagent/toxin/xeno_neurotoxin
+
+/datum/status_effect/upgrade_toxin/on_apply()
+	if(!isxeno(owner))
+		return FALSE
+	buff_owner = owner
+	chamber_scaling = length(buff_owner.hive.veil_chambers)
+	RegisterSignal(SSdcs, COMSIG_UPGRADE_CHAMBER_UTILITY, PROC_REF(update_buff))
+	RegisterSignal(buff_owner, COMSIG_XENOMORPH_ATTACK_LIVING, PROC_REF(on_slash))
+	return TRUE
+
+/datum/status_effect/upgrade_toxin/on_remove()
+	UnregisterSignal(SSdcs, COMSIG_UPGRADE_CHAMBER_UTILITY)
+	UnregisterSignal(buff_owner, COMSIG_XENOMORPH_ATTACK_LIVING)
+	return ..()
+
+/// Sets the chamber_scaling to the amount of active veil chambers.
+/datum/status_effect/upgrade_toxin/proc/update_buff()
+	SIGNAL_HANDLER
+	chamber_scaling = length(buff_owner.hive.veil_chambers)
+
+/// Injects the human target with a variable amount of the selected reagent.
+/datum/status_effect/upgrade_toxin/proc/on_slash(datum/source, mob/living/target)
+	SIGNAL_HANDLER
+	if(!chamber_scaling || target.stat == DEAD || !ishuman(target) || !target.can_sting())
+		return
+	var/mob/living/carbon/human/human_target = target
+	human_target.reagents.add_reagent(selected_reagent, toxin_amount_per_chamber * chamber_scaling)
+
+// ***************************************
+// ***************************************
+// ***************************************
+/atom/movable/screen/alert/status_effect/upgrade_pheromones
+	name = "Pheromones"
+	desc = "Allows to emit pheromones."
+	icon_state = "xenobuff_phero"
+
+/atom/movable/screen/alert/status_effect/upgrade_pheromones/Click()
```

Comment:
Needs to be more obvious that this is how you change what pheremones you're using, might be enough to have the description mention this for pheremones and slash chemicals